c7n/resources/securityhub.py

```diff
@@ -83,6 +88,103 @@ def register_resources(klass, registry, resource_class):
         resource_class.filter_registry.register('finding', klass)
 
 
+def get_security_hub(session_factory, retry):
+
+    client = local_session(session_factory).client('securityhub')
+    hub = retry(client.describe_hub, ignore_err_codes=(
+        'InvalidAccessException',))
+    if not hub:
+        return []
+
+    hub.pop('ResponseMetadata')
+    return [hub]
+
+
+@resources.register('security-hub')
+class SecurityHubMember(QueryResourceManager):
+
+    filter_registry = filters
+    action_registry = actions
+
+    class resource_type(TypeInfo):
+        service = 'securityhub'
+        id = name = 'HubArn'
+        arn_type = 'hub'
+        cfn_type = 'AWS::SecurityHub::Hub'
+        enum_spec = ("describe_hub", None, None)
+
+    def resources(self):
+        return self.filter_resources(get_security_hub(self.session_factory, self.retry))
+
+    def get_resources(self, resource_ids):
+        return get_security_hub(self.session_factory, self.retry)
+
+
+@filters.register('valid-master')
+class SecurityHubValidMasterFilter(Filter):
+    """Filter SecurityHub by checking if it has a valid master account
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: security-hub-valid-master
+            resource: security-hub
+            filters:
+              - type: valid-master
+                accounts:
+                  - "123456789"
+    """
+    schema = type_schema('valid-master', accounts={'type': 'array', 'items': {'type': 'string'}})
+    permissions = ("securityhub:GetMasterAccount",)
+
+    def process(self, balancers, event=None):
+        client = local_session(self.manager.session_factory).client('securityhub')
+        master_account_resp = client.get_master_account()
+        master_account_id = master_account_resp.get('Master', {}).get("AccountId")
+        if master_account_id in self.data["accounts"]:
+            return balancers
+        else:
+            return []
+
+
+@filters.register('enabled-products')
```

Comment:
s/enabled-products/products

so.. in a multi account setup.. afaicr products need to be setup at the master level.. ie not operable on via target.